src/geoFire/interfaces/geoQueryState.ts

```diff
@@ -0,0 +1,9 @@
+import * as firebase from 'firebase';
+
+export interface GeoQueryState {
+  active: boolean;
+  childAddedCallback: (a: firebase.database.DataSnapshot, b?: string) => any;
```

Comment:
It seems like we could not repeat ourselves here with the callback signature. No?

Also, could we be more descriptive with the prop names than `a` and `b`?

Reply:
While it may not be necessary, I'm not sure what a better/ideal solution would be. Effectively I can change it to whatever you thing is best. I'm just not sure what it would be off of the top of my head.

Reply:
Lets just create a type and assign each of these to that?

```typescript
export type GeoQueryStateCallback = (snap: firebase.database.DataSnapshot, ) => any;

export interface GeoQueryState {
  childAddedCallback: GeoQueryStateCallback;
  ...
}
```

As far as the property names go, I just looked at our .d.ts for the typescript package and we use the same variable names so I guess we are probably fine with that.